frontend/src/pages/BYONImages/AddNewImageForm.tsx

```diff
@@ -0,0 +1,296 @@
+        // TODO create validator for create form
+        break;
+      default:
+        // when no option is selected
+        valid = false;
+    }
+
+    return valid;
+  }, [formState.state, imageSelectValue]);
+
+  // set react component form based on selection
+  const imageSelectForm = useMemo(() => {
+    switch (imageSelectValue) {
+      case 'import':
+        return (
+          <ImportImageForm
+            state={formState.state}
+            valid={formState.valid}
+            setValue={(key, value) => formDispatch({ type: 'set', key: key, value: value })}
+          />
+        );
+      case 'create':
+        // TODO create the Create page
+        return null;
+      case 'build':
+        // TODO create the Build page
+        return null;
+      default:
+        return null;
+    }
+  }, [imageSelectValue, formState]);
+
+  // when canceling the form
+  const closeHandler = () => {
+    navigate('/notebookImages');
+  };
+
+  // when the form is submitted, send data to api
+  const onSubmit = () => {
+    // TODO add support for remote validation
+    // this currently only runs a local validation
+    // i.e. let the user submit after local validation,
+    // but hold the api call until remote validation is successful
+
+    // do not submit if not valid
+    if (!isValid) {
+      return;
+    }
+
+    setIsLoading(true);
+
+    // submit form based on which form is selected
+    switch (imageSelectValue) {
+      case 'import':
+        importBYONImage({
+          name: formState.state.name,
+          url: formState.state.repository,
+          description: formState.state.description,
+          user: userName,
+          software: formState.state.software,
+          packages: formState.state.packages,
+        })
+          .then((value: ResponseStatus) => {
+            if (value.success === false) {
+              dispatch(
+                addNotification({
+                  status: 'danger',
+                  title: 'Error',
+                  message: `Unable to add notebook image ${name}`,
+                  timestamp: new Date(),
+                }),
+              );
+            }
+            forceUpdate();
+            closeHandler();
+          })
+          .catch((error) => {
+            setIsLoading(false);
+            setError(error.toString());
+          });
+        break;
+      case 'create':
+        // TODO create validator and submitter
+        break;
+      case 'build':
+        // TODO create the Build page
+        break;
+    }
+  };
+
+  return (
+    <React.Fragment>
+      <Breadcrumb className="breadcrumb">
+        <BreadcrumbItem>Settings</BreadcrumbItem>
+        <BreadcrumbItem to="/notebookImages">Notebook image settings</BreadcrumbItem>
+        <BreadcrumbItem>Add new image</BreadcrumbItem>
+      </Breadcrumb>
+      <ApplicationsPage
+        loaded
+        empty={false}
+        title="Add new image"
+        description="Add a new image to Red Hat Data Science
```

Comment:
I believe this should be:
```suggestion
        description="Add a new image to Open Data Hub
```

but beyond that: is there a better place to set the product name globally? (thinking in moving this downstream later)

Reply:
that is the only place red hat data science is mentioned. Although we could set some global variable, it is not needed if it isnt used much